src/hardware/voodoo.cpp

```diff
@@ -6996,6 +7046,45 @@ static constexpr uint32_t voodoo_r(const uint32_t addr)
 	return 0xffffffff;
 }
 
+int get_cpu_core_count()
+{
+	// Define reasonable limits
+	constexpr auto MinCount     = 1;
+	constexpr auto DefaultCount = 4;
+	constexpr auto MaxCount     = 1024;
+
+	auto is_valid = [](const int count) {
+		return count >= MinCount && count <= MaxCount;
+	};
+
+	auto count = SDL_GetCPUCount();
+	if (is_valid(count)) {
+		return count;
+	}
+
+	count = static_cast<int>(std::thread::hardware_concurrency());
+	if (is_valid(count)) {
+		return count;
+	}
+
+	return DefaultCount;
+}
+
+// Sweet spot number of Voodoo threads came from these benchmarks:
+// https://github.com/schellingb/dosbox-pure/issues/300#issuecomment-2144050898
+//
+static int get_optimal_num_threads()
+{
+	constexpr auto MinNumThreads = 1;
+	constexpr auto MaxNumThreads = 7;
+
+	static auto num_threads = std::clamp(get_cpu_core_count() - 1,
+	                                     MinNumThreads,
+	                                     MaxNumThreads);
+
+	return num_threads;
+}
```

Comment:
Any reason for the high levels of mistrust of these CPU core count query function? 🤣 

I expect `SDL_GetCPUCount` to be battle-tested and always work correctly. Besides, you clamp to 1 to 7 anyway, so this validation and fallback mechanism is pointless.

I strongly suggest to drop this function and use `SDL_GetCPUCount` directly wherever you need the core count.

Reply:
I agree; now simplified and added a couple comments in the function.

Reply:
Yeah, I highly doubt they'd ever actually fail in practice 😉